OpenVPN/openvpn/ingest/parser.yml

```
@@ -65,3 +87,45 @@ stages:
           observer.mac: "{{parsed_event.message.hwadr}}"
           observer.ip: "{{parsed_event.message.observer_ip}}"
           openvpn.peer.ip: "{{parsed_event.message.peer_ip}}"
+
+      - set:
+          user.name: "{{parsed_event.message.user_name}}"
+
+      - set:
+          user.name: "{{parse_ldap_kv.message.CN}}"
+        filter: "{{ parse_ldap_kv.message != null }}"
+
+      - set:
+          # DC key is repeated (e.g. `DC=example,DC=org`), so we have to extract and concatenate values manually
+          source.domain: '{{ parsed_event.message.ldap_kvs[parsed_event.message.ldap_kvs.index("DC=") + 3:].replace(",DC=", ".") }}'
```

This information represents the domain the user belong to.
```suggestion
          user.domain: '{{ parsed_event.message.ldap_kvs[parsed_event.message.ldap_kvs.index("DC=") + 3:].replace(",DC=", ".") }}'
```